src/modules/statement/comment_doc.rs

```diff
@@ -36,16 +37,35 @@ impl SyntaxModule<ParserMetadata> for CommentDoc {
                             // Update the column of the last comment
                             col = token.pos.0;
                             meta.increment_index();
-                            // If the comment signifies a paragraph break, we add two newlines
-                            if token.word[3..].trim().is_empty() {
-                                if last_char == '\n' {
-                                    continue;
+
+                            let line = &token.word[3..];
+                            let trimmed_line = &line.trim();
+
+                            if trimmed_line.is_empty() {
+                                // If the comment signifies a paragraph break, we add two newlines
+                                if last_char != '\n' {
+                                    self.value.push_str("\n\n");
+                                }
+                            } else {
+                                if last_char != '\n' { self.value.push(' ') }
+                                match *trimmed_line {
+                                    "```ab" => {
+                                        inside_code_block = true;
+                                        self.value.push_str(trimmed_line);
+                                    },
+                                    "```" => {
+                                        inside_code_block = false;
+                                        self.value.push_str(trimmed_line);
+                                    },
+                                    // Add code lines without trimming the start
+                                    _ if inside_code_block => {
+                                        self.value.push_str(line[1..].trim_end());
+                                    },
```

Comment:
This probably assumes that each codeblock is one space away. This example:
```ab
///```ab
///let a = 12
///```
```
Generates:
````md
```ab
et a = 12
```
````
I think that we should probably `left_trim` by the same amount we `left_trim` the starting block (` ```ab `).